ksql-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -171,15 +173,27 @@ public KsqlExecutionContext createSandbox() {
   public ExecuteResult execute(
       final ConfiguredStatement<?> statement
   ) {
+    if (!statement.getChecksum().matches(checksum)) {
```

Comment:
It would be good to log the current checksum, the expected checksum, and the statement at info level here (before checking the checksum).